### PR TITLE
Feature scipy differential evolution optimizer from develop

### PR DESCRIPTION
**Pull request by PhilippStaedter**
second try to merge the implemented optimizer scipy.optimize.differential_evolution for pypesto into develop. Therefore, it has been branched from develop instead of master to hopefully avoid merge conflicts

### CODE REVIEW

File: pypesto/optimize/optimizer.py [hunk 1]
```diff
@@ -662,6 +666,61 @@ def is_least_squares(self):
...
+class ScipyDifferentialEvolutionOptimizer(Optimizer):
+    """
+    Global optimization using scipy's differential evolution optimizer.
+    Package homepage: https://docs.scipy.org/doc/scipy/reference/generated\
+        /scipy.optimize.differential_evolution.html
+
+    Parameters
+    ----------
+    options:
+        Optimizer options that are directly passed on to scipys' optimizer
```
Comment (yannikschaelte) [inline pypesto/optimize/optimizer.py:678]:
```suggestion
        Optimizer options that are directly passed on to scipy's optimizer.
```

File: pypesto/optimize/optimizer.py [hunk 2]
```diff
@@ -662,6 +666,61 @@ def is_least_squares(self):
...
+
+
+    Examples
+    --------
+    Arguments that can be passed to options:
+
+    maxiter:
+        used to calculate the maximal number of funcion evaluations by
+        maxfevals = (maxiter + 1) * popsize * len(x)
```
Comment (yannikschaelte) [inline pypesto/optimize/optimizer.py:687]:
```suggestion
        maxfevals = (maxiter + 1) * popsize * len(x)
        Default: 100
```

File: pypesto/optimize/optimizer.py [hunk 3]
```diff
@@ -662,6 +666,61 @@ def is_least_squares(self):
...
+    popsize:
+        population size, default value 15
+    """
+
+    def __init__(self, options: Dict = None):
+        super().__init__()
+
+        if options is None:
+            options = {'maxiter': 66}
```
Comment (yannikschaelte) [inline pypesto/optimize/optimizer.py:696]:
```suggestion
            options = {'maxiter': 100}
```
I agree that the scipy default of 1000 (--> neval = 1000 * 15 * len(x)) is too long. Maybe this is a good default. 66 is a bit arbitrary.

**Review by yannikschaelte — APPROVED** [review]
Please shortly address the minor changes, then good to go from my side.

**Comment by yannikschaelte** [comment]
Oh, and please add the optimizer to the tests in https://github.com/ICB-DCM/pyPESTO/blob/develop/test/optimize/test_optimize.py#L24. I think that got lost on the way somewhere.

File: pypesto/optimize/optimizer.py (hunk 1 above)
Comment (PhilippStaedter) [reply pypesto/optimize/optimizer.py]: Done

File: pypesto/optimize/optimizer.py (hunk 2 above)
Comment (PhilippStaedter) [reply pypesto/optimize/optimizer.py:687]: Done

File: pypesto/optimize/optimizer.py (hunk 3 above)
Comment (PhilippStaedter) [reply pypesto/optimize/optimizer.py]: Done

**Review by LeonardSchmiester — APPROVED** [review]

**Review by paulstapor — APPROVED** [review]
Looks good, can merge, when tests are through

**Pull request merged**